src/openalea/lpy/lsysparameters.py

```diff
@@ -0,0 +1,612 @@
+            if len(value) > 0:
+                txt += indentation+key+' = '+repr(str(value))+'\n'
+        return txt
+
+    def generate_json_parameter_dict(self):
+        import openalea.plantgl.algo.jsonrep as jrep
+        from collections import OrderedDict 
+
+        parameters = OrderedDict()
+        for category in self.parameters.values():
+            panel = category.info.copy()
+            panel['name']
+            panel.setdefault('enabled',True)
+            panel['scalars'] = [scalar.todict() for scalar in category.scalars]
+            items = []
+            for manager,obj in category.items:
+                items.append(manager.to_json(obj))
+            panel['items'] = items
+            parameters[panel['name']] = panel
+
+        defaultlist = PglTurtle().getColorList()
+        materials = []
+        for i, cmat in self.colorList.items():
+            if not isSimilarToDefaultTurtleMat(cmat, i):
+                jmat = jrep.to_json_rep(cmat)
+                jmat['index'] = i
+                materials.append(jmat)
+
+        if self.animation_timestep is None:
+            options = self.execOptions
+        else:
+            options = self.execOptions.copy()
+            options['animation_timestep'] = self.animation_timestep
+
+        result =  dict(
+            schema = 'lpy',
+            version = str(default_lpyjson_version),
+            options = options,
+            materials = materials,
+            parameters = list(parameters.values()),
+            credits = dict([(key,value) for key,value in self.credits.items() if value != ''])
+        )
+        assert LsystemParameters.is_valid_schema(result)
+        return result     
+
+    @staticmethod
+    def is_valid_schema(obj):
+        # TODO: load files only once
+        import io, os, json, jsonschema
+        is_valid = False
+        schema_path = os.path.join(os.path.dirname(__file__), 'parameters', 'schema')
+        with io.open(os.path.join(schema_path, 'lpy.json'), 'r') as schema_file:
+            try:
+                schema = json.loads(schema_file.read())
+            except json.JSONDecodeError as e:
+                print(e)
+            resolver = jsonschema.RefResolver(f'file:///{schema_path}/', schema)
+            try:
+                jsonschema.validate(obj, schema, format_checker=jsonschema.draft7_format_checker, resolver=resolver)
+                is_valid = True
+            except jsonschema.exceptions.ValidationError as e:
+                print('L-Py schema validation failed:', e.message)
+            except jsonschema.exceptions.RefResolutionError as e:
+                print('JSON schema $ref file not found:', e)
+        return is_valid
+
+    def retrieve_from_json_dict(self, obj):
+        import openalea.plantgl.algo.jsonrep as jrep
+        from openalea.lpy.parameters.scalar import scalar_from_json_rep
+        assert LsystemParameters.is_valid_schema(obj)
```

Comment:
Would be nice if the error handling could be a bit more convenient:

e.g. returning a boolean instead of an AssertionError in .load & .loads

Reply:
Done in 9e4f59e6fa8c837137e31df097fec366f3a505e8